File: 06/main.cpp
#include <iostream>
#include <vector>

using namespace std;
class Solution {
public:
	int minNumberInRotateArray(vector<int> rotateArray) {
		if (rotateArray.empty())
			return 0;
		for (int i = 1; i < rotateArray.size(); i++)
			if (rotateArray[i] < rotateArray[i - 1])
				return rotateArray[i];
			return rotateArray[0];
	}
};

int main(){
	Solution so;
	vector <int> vec;

	int x;
	cin >> x;
	while (x > 0){
		vec.push_back(x);
		cin >> x;
	}

	int y = so.minNumberInRotateArray(vec);
	cout << "The number is:"<<y << endl;

	getchar(); getchar();
	return 0;
}

File: 06/main.py
# -*- coding:utf-8 -*-
class Solution:
    def __init__(self):
        pass
    
    def minNumberInRotateArray(self, rotateArray):
        if len(rotateArray)==0:
            return 0
        for i in range(1,len(rotateArray)):
            if rotateArray[i]<rotateArray[i-1]:
                return rotateArray[i]
        return rotateArray[0]
            
    
so=Solution()#()
array=[]
x=int(raw_input())
while x>0:
    array.append(x)
    x=int(raw_input())
y=so.minNumberInRotateArray(array)
print "The number is:",y

